llama_index/agent/retriever_openai_agent.py

```diff
@@ -1,96 +1,31 @@
     """Function Retriever OpenAI Agent.
 
     Uses our object retriever module to retrieve openai agent.
 
-    """
+    NOTE: This is deprecated, you can just use the base `OpenAIAgent` class by
+    specifying the following:
+    ```
+    agent = OpenAIAgent.from_tools(tool_retriever=retriever, ...)
+    ```
 
-    def __init__(
-        self,
-        retriever: ObjectRetriever[BaseTool],
-        llm: OpenAI,
-        memory: BaseMemory,
-        prefix_messages: List[ChatMessage],
-        verbose: bool = False,
-        max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
-        callback_manager: Optional[CallbackManager] = None,
-    ) -> None:
-        super().__init__(
-            llm=llm,
-            memory=memory,
-            prefix_messages=prefix_messages,
-            verbose=verbose,
-            max_function_calls=max_function_calls,
-            callback_manager=callback_manager,
-        )
-        self._retriever = retriever
+    """
 
     @classmethod
     def from_retriever(
-        cls,
-        retriever: ObjectRetriever[BaseTool],
-        llm: Optional[OpenAI] = None,
-        chat_history: Optional[List[ChatMessage]] = None,
-        memory: Optional[BaseMemory] = None,
-        memory_cls: Type[BaseMemory] = ChatMemoryBuffer,
-        verbose: bool = False,
-        max_function_calls: int = DEFAULT_MAX_FUNCTION_CALLS,
-        callback_manager: Optional[CallbackManager] = None,
-        system_prompt: Optional[str] = None,
-        prefix_messages: Optional[List[ChatMessage]] = None,
+        cls, retriever: ObjectRetriever[BaseTool], **kwargs: Any
     ) -> "FnRetrieverOpenAIAgent":
-        chat_history = chat_history or []
-        llm = llm or OpenAI(model=DEFAULT_MODEL_NAME)
-        if not isinstance(llm, OpenAI):
-            raise ValueError("llm must be a OpenAI instance")
-
-        if callback_manager is not None:
-            llm.callback_manager = callback_manager
-
-        memory = memory or memory_cls.from_defaults(chat_history=chat_history, llm=llm)
-
-        if not is_function_calling_model(llm.model):
-            raise ValueError(
-                f"Model name {llm.model} does not support function calling API. "
-            )
-
-        if system_prompt is not None:
-            if prefix_messages is not None:
-                raise ValueError(
-                    "Cannot specify both system_prompt and prefix_messages"
-                )
-            prefix_messages = [ChatMessage(content=system_prompt, role="system")]
-
-        prefix_messages = prefix_messages or []
-
-        return cls(
-            retriever=retriever,
-            llm=llm,
-            memory=memory,
-            prefix_messages=prefix_messages,
-            verbose=verbose,
-            max_function_calls=max_function_calls,
-            callback_manager=callback_manager,
+        return cast(
+            FnRetrieverOpenAIAgent, cls.from_tools(tool_retriever=retriever, **kwargs)
```

Comment:
very nit: is the only reason to keep around this class to maintain backward compatibility for this method? 

If so, we can possibly just move this method to OpenAIAgent (and raise a deprecation warning), and have `FnRetrieverOpenAIAgent` purely as an alias as well.